sys/hoon.hoon

```diff
@@ -9765,10 +9765,11 @@
     ^-  type
     ?-  -.lap
       %&  p.lap
-      %|  %-  fire
+      %|  %-  fork
           %+  turn  ~(tap in q.lap)
-          |=  {a/type b/foot}
-          [a [%dry %$ 1]]
+          |=  [a=type *]
+          ?>  ?=([%core *] a)
+          [%core q.q.a q.a]
```

Comment:
I know I should know this, but what's the difference between the type in the %core and the type in the `+coil`? How do `p.a` and `q.q.a` differ here?

Reply:
This is a good question and I was thinking about this too.  q.q is the original payload type and p is the current payload type.  Here I was again mimicking +fire too closely (re: firing a dry arm).  But I think I can just use the original type, unmodified.  Give me a second to test this...

Reply:
When I have +felt return `a` as the unmodified type I get a nest-fail on rebuild (for clay).  Something depends upon using the q.q type for wing resolution...

This issue will be revisited in the future (as is often said, the compiler needs some rewriting), so I suggest setting aside this simplification for the moment.

Reply:
Thanks -- I was just curious.